rs/artifact_pool/src/ingress_pool.rs

```diff
@@ -34,26 +38,24 @@ struct IngressPoolSection<T: AsRef<IngressPoolObject>> {
     /// associated functions [`insert`], [`remove`] and [`purge_below`].
     artifacts: BTreeMap<IngressMessageId, T>,
     metrics: PoolMetrics,
-    /// Note: The byte size is updated incrementally as a side-effect of insert, remove
+    /// Note: `peer_counters` is updated incrementally as a side-effect of insert, remove
     /// and purge invocations. Never modify the artifacts map directly! Use the
     /// associated functions [`insert`], [`remove`] and [`purge_below`]
-    byte_size: usize,
+    peer_counters: PeerCounters,
 }
 
-impl<T: AsRef<IngressPoolObject>> CountBytes for IngressPoolSection<T> {
-    fn count_bytes(&self) -> usize {
-        self.byte_size
-    }
-}
 impl<T: AsRef<IngressPoolObject>> IngressPoolSection<T> {
-    fn new(metrics: PoolMetrics) -> IngressPoolSection<T> {
+    fn new(log: ReplicaLogger, metrics: PoolMetrics) -> IngressPoolSection<T> {
         IngressPoolSection {
+            peer_counters: PeerCounters::new(log),
             artifacts: BTreeMap::new(),
             metrics,
-            byte_size: 0,
         }
     }
 
+    /// Adds the `artifact` to the pool.
+    /// Note: the function doesn't check if the pool's limits are already exceeded, it's the
+    /// caller's responsibility to check the limits before calling `insert`.
```

Comment:
should we add to this comment that this check happens in the on_state_change of the ingress_handler.rs?

Reply:
Done